homeassistant/components/plex/config_flow.py

```diff
@@ -0,0 +1,216 @@
+                    )
+
+            server_id = plex_server.machine_identifier
+
+            for entry in self._async_current_entries():
+                if entry.data[CONF_SERVER_IDENTIFIER] == server_id:
+                    return self.async_abort(reason="already_configured")
+
+            url = plex_server.url_in_use
+            token = user_input.get(CONF_TOKEN)
+
+            server_config = {CONF_URL: url}
+            if token:
+                server_config[CONF_TOKEN] = token
+            if url.startswith("https"):
+                server_config[CONF_VERIFY_SSL] = user_input.get(
+                    CONF_VERIFY_SSL, DEFAULT_VERIFY_SSL
+                )
+
+            _LOGGER.debug("Valid config created for %s", plex_server.friendly_name)
+
+            return self.async_create_entry(
+                title=plex_server.friendly_name,
+                data={
+                    CONF_SERVER: plex_server.friendly_name,
+                    CONF_SERVER_IDENTIFIER: server_id,
+                    PLEX_SERVER_CONFIG: server_config,
+                },
+            )
+
+        return self.async_show_form(
+            step_id="user", data_schema=data_schema, errors=errors
+        )
+
+    async def async_step_manual_setup(self, user_input=None):
+        """Begin manual configuration."""
+        if user_input is None:
+            data_schema = vol.Schema(
+                {
+                    vol.Required(
+                        CONF_HOST, default=self.discovery_info.get(CONF_HOST)
+                    ): str,
+                    vol.Required(
+                        CONF_PORT,
+                        default=int(self.discovery_info.get(CONF_PORT, DEFAULT_PORT)),
+                    ): int,
+                    vol.Optional(CONF_SSL, default=DEFAULT_SSL): bool,
+                    vol.Optional(CONF_VERIFY_SSL, default=DEFAULT_VERIFY_SSL): bool,
+                    vol.Optional(CONF_TOKEN): str,
+                }
+            )
+            return self.async_show_form(step_id="manual_setup", data_schema=data_schema)
+
+        host = user_input.pop(CONF_HOST)
+        port = user_input.pop(CONF_PORT)
+        prefix = "https" if user_input.get(CONF_SSL) else "http"
+        user_input[CONF_URL] = f"{prefix}://{host}:{port}"
+        return await self.async_step_user(user_input=user_input)
+
+    async def async_step_select_server(self, user_input=None):
+        """Use selected Plex server."""
+        if user_input is None:
+            return self.async_show_form(
+                step_id="select_server",
+                data_schema=vol.Schema(
+                    {vol.Required(CONF_SERVER): vol.In(self.available_servers)}
+                ),
+                errors={},
+            )
+
+        config = self.current_login
+        config[CONF_SERVER] = user_input.get(CONF_SERVER)
+        return await self.async_step_user(user_input=config)
```

Comment:
I think we need a separate step for creating the server and checking connection. The user input of the user step form doesn't match what we send here or from the import step. The form schema for each step should match the user input.

Reply:
The dict keys don't match, but they're both valid inputs as `user_input`. Look at this comment for some ways it can be configured: https://github.com/home-assistant/home-assistant/pull/26548#discussion_r322983036

Reply:
The user step should only accept the token as input. We should create another step for the server config input.

Reply:
The server create step method doesn't have to accept the server config as input, as the method won't have a form. The method could also take the config from `self.current_login`.